bigquery/datasets.js

```diff
@@ -0,0 +1,201 @@
+ */
+function createDataset (name, projectId, callback) {
+  // Instantiate a bigquery client
+  var bigquery = BigQuery({
+    projectId: projectId
+  });
+  var dataset = bigquery.dataset(name);
+
+  // See https://googlecloudplatform.github.io/gcloud-node/#/docs/bigquery/latest/bigquery
+  dataset.create(function (err, dataset) {
+    if (err) {
+      return callback(err);
+    }
+
+    console.log('Created dataset: %s', name);
+    return callback(null, dataset);
+  });
+}
+// [END create_dataset]
+
+// [START delete_dataset]
+/**
+ * List datasets in the authenticated project.
+ *
+ * @param {string} name The name for the new dataset.
+ * @param {string} projectId The project ID to use.
+ * @param {function} callback The callback function.
+ */
+function deleteDataset (name, projectId, callback) {
+  // Instantiate a bigquery client
+  var bigquery = BigQuery({
+    projectId: projectId
+  });
+  var dataset = bigquery.dataset(name);
+
+  // See https://googlecloudplatform.github.io/gcloud-node/#/docs/bigquery/latest/bigquery
+  dataset.delete(function (err) {
+    if (err) {
+      return callback(err);
+    }
+
+    console.log('Deleted dataset: %s', name);
+    return callback(null);
+  });
+}
+// [END delete_dataset]
+
+// [START list_datasets]
+/**
+ * List datasets in the authenticated project.
+ *
+ * @param {string} projectId The project ID to use.
+ * @param {function} callback The callback function.
+ */
+function listDatasets (projectId, callback) {
+  // Instantiate a bigquery client
+  var bigquery = BigQuery({
+    projectId: projectId
+  });
+
+  // See https://googlecloudplatform.github.io/gcloud-node/#/docs/bigquery/latest/bigquery
+  bigquery.getDatasets(function (err, datasets) {
+    if (err) {
+      return callback(err);
+    }
+
+    console.log('Found %d dataset(s)!', datasets.length);
+    return callback(null, datasets);
+  });
+}
+// [END list_datasets]
+
+// [START get_dataset_size]
+/**
+ * Calculate the size of the specified dataset.
+ *
+ * @param {string} datasetId The ID of the dataset.
+ * @param {string} projectId The project ID of the dataset.
+ * @param {function} callback The callback function.
+ */
+function getDatasetSize (datasetId, projectId, callback) {
+  // Instantiate a bigquery client
+  var bigquery = BigQuery({
+    projectId: projectId
+  });
+  var dataset = bigquery.dataset(datasetId);
+
+  // See https://googlecloudplatform.github.io/gcloud-node/#/docs/bigquery/latest/bigquery/dataset
+  dataset.getTables(function (err, tables) {
+    if (err) {
+      return callback(err);
+    }
+
+    return async.map(tables, function (table, cb) {
+      // Fetch more detailed info for each table
+      table.get(function (err, tableInfo) {
+        if (err) {
+          return cb(err);
+        }
+        // Return numBytes converted to Megabytes
+        var numBytes = tableInfo.metadata.numBytes;
+        return cb(null, (parseInt(numBytes, 10) / 1000) / 1000);
```

Comment:
I think it's much easier for the user if we just display in MB, saving them the mental overhead.